### PR TITLE
Connection management

### PR DESCRIPTION
**Pull request by lkrawczyk**

### CODE REVIEW

File: dev/src/base/net/src/connectionManager.cpp [hunk 1]
```diff
@@ -7,42 +7,358 @@
...
+
+            return 0;
+        }
+
+        CAN_YIELD Int32 ConnectionManager::sendRaw(const void* data, Uint32 size, const Address& destination)
+        {
+            Int32 bytesSent = 0;
+            do
+            {
+                bytesSent = send(data, size, destination);
```
Comment (rexdex) [inline dev/src/base/net/src/connectionManager.cpp]: Is the data pointer moved somehow ?

File: dev/src/base/net/src/connectionManager.cpp [hunk 2]
```diff
@@ -7,42 +7,358 @@
...
+                acceptInfo.m_datagram = receiveBuffer;
+                acceptInfo.m_datagramSize = receivedSize;
+                acceptInfo.m_sourceAddress = sourceAddress;
+
+                m_acceptQueue.push(acceptInfo);
+
+                return;
+            }
+            break;
+            default:
```
Comment (rexdex) [inline dev/src/base/net/src/connectionManager.cpp]: If possible I would make all packet types explicit: ie. Connect, Data, etc. the default case should than be a "unknown packet type" kind of thing.

File: dev/src/base/net/src/connectionManager_test.cpp [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    server.m_func = [this](Uint64 i) {
+        auto connectionId = m_server.accept();
+        ASSERT_NE(connectionId, INVALID_ID);
+    };
+
+    auto clientCounter = Fibers::Schedule(client);
+    auto serverCounter = Fibers::Schedule(server);
+
+    Fibers::WaitForCounterAndRelease(clientCounter);
+    Fibers::WaitForCounterAndRelease(serverCounter);
```
Comment (rexdex) [inline dev/src/base/net/src/connectionManager_test.cpp]: Would be good to check that the accept happens after connection was requested

File: dev/src/base/net/src/connectionManager_test.cpp [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    };
+
+    fibers::Job server;
+    server.m_func = [this](Uint64 i)
+    {
+        auto connectionId = m_server.accept();
+        ASSERT_NE(connectionId, INVALID_ID);
+
+        auto receiveResult = m_server.receive(connectionId, m_packetAllocator);
+        ASSERT_EQ(receiveResult.m_status, net::ConnectionStatus::Connected);
```
Comment (rexdex) [inline dev/src/base/net/src/connectionManager_test.cpp]: Packet content is not checked. Word of KEK should never be lost in transit.

File: dev/src/base/net/src/connectionManager.cpp (hunk 1 above)
Comment (lkrawczyk) [reply dev/src/base/net/src/connectionManager.cpp]: It is internally copied by send() -> sendto() call chain, the ownership of the original data buffer doesn't change. Can this blow up? I assume the lifespan of the data should be at least the same as the period when sendRaw() blocks the fiber.

File: dev/src/base/net/src/connectionManager.cpp (hunk 2 above)
Comment (lkrawczyk) [reply dev/src/base/net/src/connectionManager.cpp]: Yes, and it should be immediately dropped, because it is invalid from our standpoint.

**Pull request merged**